smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/RequestCompressionTraitValidator.java

```diff
@@ -0,0 +1,102 @@
+            validateEncodings(model, shape, events);
+        });
+        return events;
+    }
+
+    private void validateOperationInput(Model model, OperationShape operationShape, List<ValidationEvent> events) {
+        // Validate operation input has no member with both streaming and requiresLength
+        // traits applied
+        if (operationShape.getInputShape().equals(ShapeId.from("smithy.api#Unit"))) {
+            events.add(error(operationShape,
+                    "The requestCompression trait can only be applied to operations which have an "
+                            + "input shape defined"));
+            return;
+        }
+        StructureShape inputShape = model.expectShape(operationShape.getInputShape(), StructureShape.class);
+        for (Map.Entry<String, MemberShape> entry : inputShape.getAllMembers().entrySet()) {
+            MemberShape memberShape = entry.getValue();
+            Shape targetShape = model.expectShape(memberShape.getTarget());
+            if (targetShape.hasTrait(StreamingTrait.class)
+                    && targetShape.hasTrait(RequiresLengthTrait.class)) {
+                events.add(error(operationShape, String.format(
+                        "The requestCompression trait can only be applied to operations which do not have "
+                                + "input members that target shapes with both the streaming and requiresLength"
+                                + "trait applied, but found member `%s` targeting `%s`",
+                        memberShape.getId(), targetShape.getId())));
+            }
+        }
+    }
+
+    private void validateEncodings(Model model, OperationShape operationShape, List<ValidationEvent> events) {
+        // Validate encodings have at least one compression algorithm
+        RequestCompressionTrait trait = operationShape.expectTrait(RequestCompressionTrait.class);
+        if (trait.getEncodings().isEmpty()) {
+            events.add(error(operationShape, trait, String.format(
+                    "there must be at least one compression algorithm in `encodings` for the "
+                            + "requestCompression trait applied to `%s`",
+                    operationShape.getId())));
+            return;
+        }
+        // Validate encodings are all supported compression algorithms
+        List<String> invalidEncodings = new ArrayList<>();
+        for (String encoding : trait.getEncodings()) {
+            if (!RequestCompressionTrait.SUPPORTED_COMPRESSION_ALGORITHMS.contains(encoding.toLowerCase())) {
+                invalidEncodings.add(encoding);
+            }
+        }
+        if (!invalidEncodings.isEmpty()) {
+            events.add(error(operationShape, trait, String.format(
+                    "Invalid compression algorithms found in requestCompression trait applied to `%s`: %s",
+                    operationShape.getId(),
+                    String.join(", ", invalidEncodings))));
```

Comment:
Use plural when needed and add ticks to each of the invalid encodings.

```suggestion
            events.add(error(operationShape, trait, String.format(
                    "Invalid compression algorithm%s found in requestCompression trait applied to `%s`: %s",
                    operationShape.getId(),
                    (invalidEncodings.size() == 1 ? "" : "s"),
                    ValidationUtils.tickedList(invalidEncodings))));
```

Reply:
I think this would need to swap the first and second parameters to `format`.